api/api.go

```diff
@@ -461,6 +523,159 @@ func CreateOrb(ctx context.Context, logger *logger.Logger, namespace string, nam
 	return orb, err
 }
 
+func incOrbVersion(orbVersion string, segment string) (string, error) {
+	v, err := semver.NewVersion(orbVersion)
+	if err != nil {
+		return "", err
+	}
+
+	var v2 semver.Version
+	switch segment {
+	case "major":
+		v2 = v.IncMajor()
+	case "minor":
+		v2 = v.IncMinor()
+	case "patch":
+		v2 = v.IncPatch()
+	}
+
+	return v2.String(), nil
+}
+
+// IncrementOrb accepts an orb and segment to increment the orb.
+func IncrementOrb(ctx context.Context, logger *logger.Logger, configPath string, namespace string, orb string, segment string) (string, error) {
+	name := namespace + "/" + orb
+	id, err := getOrbID(ctx, logger, name)
+	if err != nil {
+		return "", err
+	}
+
+	v, err := OrbVersion(ctx, logger, namespace, orb)
+	if err != nil {
+		return "", err
+	}
+
+	v2, err := incOrbVersion(v, segment)
+	if err != nil {
+		return "", err
+	}
+
+	response, err := OrbPublishID(ctx, logger, configPath, id, v2)
+	if err != nil {
+		return "", err
+	}
+
+	if len(response.Errors) > 0 {
+		return "", response.ToError()
+	}
+
+	logger.Debug("Bumped %s#%s from %s by %s to %s\n.", name, id, v, segment, v2)
+
+	return v2, nil
+}
+
+// OrbVersion finds the latest published version of an orb and returns it.
+func OrbVersion(ctx context.Context, logger *logger.Logger, namespace string, orb string) (string, error) {
+	name := namespace + "/" + orb
+
+	var response struct {
+		Orb struct {
+			Versions []struct {
+				Version string
+			}
+		}
+	}
+
+	query := `query($name: String!) {
+			    orb(name: $name) {
+			      versions(count: 1) {
+				    version
+			      }
+			    }
+		      }`
+
+	request := client.NewAuthorizedRequest(viper.GetString("token"), query)
+	request.Var("name", name)
+
+	address, err := GraphQLServerAddress(EnvEndpointHost())
+	if err != nil {
+		return "", err
+	}
+	graphQLclient := client.NewClient(address, logger)
+
+	err = graphQLclient.Run(ctx, request, &response)
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(response.Orb.Versions) != 1 {
+		return "", fmt.Errorf("the %s orb has never published a revision", name)
+	}
+
+	return response.Orb.Versions[0].Version, nil
+}
+
+// PromoteOrb takes an orb and a development version and increments a semantic release with the given segment.
+func PromoteOrb(ctx context.Context, logger *logger.Logger, namespace string, orb string, label string, segment string) (*PromoteOrbResponse, error) {
```

Comment:
To be consistent with other names: `OrbPromote` ?

Reply:
All of the APIs seem to use `verbNoun` for these mutations :thinking: